From d488ae9f7750ea6f984cbfe02c2ff7c9f5ca4bf0 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:50:37 -0400
Subject: [tools] lib/traceevent: Get rid of malloc_or_die() in read_token()

Message-id: <1408953210-26343-150-git-send-email-jolsa@redhat.com>
Patchwork-id: 88841
O-Subject: [PATCH RHEL7.1 BZ1133083 149/322] tools lib traceevent: Get rid of malloc_or_die() in read_token()
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 91dfa49bdd8ef9600d850ef68ec892eb70824e3d
Author: Namhyung Kim <namhyung@kernel.org>
Date: Thu Dec 12 16:36:07 2013 +0900
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/parse-filter.c b/tools/lib/traceevent/parse-filter.c
index ab9cefe..246ee81 100644
--- a/tools/lib/traceevent/parse-filter.c
+++ b/tools/lib/traceevent/parse-filter.c
@@ -109,7 +109,11 @@ static enum event_type read_token(char **tok)
      (strcmp(token, "=") == 0 || strcmp(token, "!") == 0) &&
      pevent_peek_char() == '~') {
   /* append it */
-  *tok = malloc_or_die(3);
+  *tok = malloc(3);
+  if (*tok == NULL) {
+   free_token(token);
+   return EVENT_ERROR;
+  }
   sprintf(*tok, "%c%c", *token, '~');
   free_token(token);
   /* Now remove the '~' from the buffer */
@@ -1123,6 +1127,8 @@ process_filter(struct event_format *event, struct filter_arg **parg,
    break;
   case EVENT_NONE:
    break;
+  case EVENT_ERROR:
+   goto fail_alloc;
   default:
    goto fail_print;
   }
-- 
1.7.1